test/unit/lodash.spec.js

```diff
@@ -0,0 +1,522 @@
+
+                expect(params).to.have.length(2);
+            });
+
+            it('should divide two numbers a and b', function () {
+                expect(_.divide.apply(_, params)).to.eql(Infinity);
+            });
+
+        });
+
+        describe('floor', function () {
+
+            var params;
+            var capacity;
+
+            beforeEach(function () {
+                params = datasets.floor();
+                capacity = params[0];
+            });
+
+            it('should verify params', function () {
+                expect(capacity instanceof Function).to.eql(true);
+            });
+
+            it('should return capacity of cuboid', function () {
+                expect(capacity(2, 2, 2)).to.eql(8);
+                expect(capacity(2.3, 2.25, 5.43)).to.eql(28.10025);
+                expect(capacity(-1, 2.25, 5.43)).to.eql(0);
+                expect(capacity(3, -2, 5.43)).to.eql(0);
+                expect(capacity(3, 2, -3)).to.eql(0);
+            });
+
+            it('should return number round down', function () {
+                expect(_.floor.apply(_, [capacity(2,2,2)])).to.eql(8);
+                expect(_.floor.apply(_, [capacity(2.2,3.3,4.4), 2])).to.eql(31.94);
+            });
+        });
+
+        describe('max', function () {
+
+            var params;
+
+            beforeEach(function () {
+                params = datasets.max();
+            });
+
+            it('should find max value (5) in user array', function () {
+                expect(params).to.have.length(1);
+
+                for(var i=0; i<params[0].length; i++){
+                    expect(typeof params[0][i]).to.eql('number');
+                }
+
+                expect(_.max.apply(_, datasets.max(params))).to.eql(5);
+            });
+        });
+
+
+        describe('maxBy', function () {
+
+            describe('maxBy - array', function () {
+
+                var params, arr1, arr2;
+
+                beforeEach(function () {
+                    params = datasets.maxBy1();
+                    arr1 = [6, 5, 4, 8, 3];
+                    arr2 = [-4, -2, -6, 0];
+                });
+                
+                it('should verify params', function () {
+                    expect(params).to.have.length(1);
+                    expect(arr1 instanceof Array && arr2 instanceof Array).to.eql(true);
+                });
+
+                it('should return maxBy for array', function () {
+                    expect(_.maxBy.apply(_, datasets.maxBy1(arr1))).to.eql(8);
+                    expect(_.maxBy.apply(_, datasets.maxBy1(arr2))).to.eql(0);
+                });
+            });
+
+            /*
+            describe('maxBy - function', function () {
+                it('should use maxBy for function', function () {
+
+                });
+            });
+            //I have no idea how to do it. Probably maxBy doesn't support function. Help!
+            */
+
+            describe('maxBy - object', function () {
```

Comment:
Ok, here is the thing. I think that you got it wrong:

MaxBy function from lodash is accepting ONLY an array of something. What you are supposed to test cases for are the iteratees in that case. Of course, to make exercise more entertaining you can think of some intresting contents of the array to show the point of it. However, by saying that it is "object" case for maxBy I immediately assumed that you want to test object iteratee.